aicsimageio/readers/reader.py

```diff
@@ -834,6 +834,264 @@ def mosaic_tile_dims(self) -> Optional[Dimensions]:
 
         return None
 
+    def _get_stack(
+        self,
+        mode: str,
+        drop_non_matching_scenes: bool = False,
+        select_scenes: Optional[
+            Union[List[Union[str, int]], Tuple[Union[str, int], ...]]
+        ] = None,
+        scene_character: str = "S",
+    ) -> types.MetaArrayLike:
+        """
+        Stack each scene contained in the reader into a
+        single array. This method handles the logic of determining which
+        stack function to use (dask or numpy) and whether or not to return a
+        labelled array (xr.DataArray). Users should prefer
+        to use one of get_stack, get_dask_stack, get_xarray_stack, or
+        get_xarray_dask_stack.
+
+        Parameters
+        ----------
+        mode: str
+            String describing the style of data to return. Should be one of:
+            "data", "dask_data", "xarray_data", "xarray_dask_data".
+        drop_non_matching_scenes: bool
+            During the scene iteration process, if the next scene to be added
+            to the stack has different shape or dtype, should it be dropped or
+            raise an error.  Default: False (raise an error)
+        select_scenes: Optional[
+                Union[List[Union[str, int]], Tuple[Union[str, int], ...]]]
+            Which scenes to stack into a single array. Scenes can be provided
+            as a list or tuple of scene indices or names.
+            Default: None (stack all scenes)
+
+        """
+        mode_check = ["data", "dask_data", "xarray_data", "xarray_dask_data"]
+        if mode not in mode_check:
+            raise ValueError(
+                f"Invalid mode kwarg. Found {mode} but should be one of:"
+                f"{', '.join(mode_check)}."
+            )
+        scene_stacks = []
+        scene_names = []
+
+        if select_scenes is None:
+            select_scenes = self.scenes
+
+        for i, s in enumerate(select_scenes):
+            self.set_scene(s)
+            data = getattr(self, mode)
```

Comment:
This will unfortunately cause a bug in a lot of cases. Regardless of if the scenes are passed in with `select_scenes` or inferred as all scenes (from `None`) -- we should be using the scene index and not the string name of the scene. `set_scene` simply does an index lookup in all scenes if provided a string and it's pretty common to have multiple scenes with the same name unfortunately. Int / index is just safer.

Reply:
I'm not sure I have a great sense of what the best thing to do is but happy to make any changes you want. My question is if `select_scenes` is passed as e.g. `list[str]` how do we determine which index those strings correspond to (without reinventing the wheel of what `set_scene` does? Would the idea be to check whether `select_scenes` is ints or None and if so use the integer scene index and otherwise fall back to set_scene?

Final thought that this raises is that I think our type hint for `select_scenes` allows for a mixed list/tuple of ints and strings so i guess we would have to check the type of each element of `select_scenes` inside the loop.

Reply:
Sorry my comment is all over the place.

What I really mean is that the only case that needs to be updated here is the `None` case.
In case of none, use a `range(0, len(self.scenes))` to generate the scene index to select.

In the case that the user provides a list of scenes to stack then exactly as you are, use that. Maybe add documentation about "it is recommended to use the scene integer index instead of the scene name to avoid duplicate scene name lookup issues"?

Reply:
`select_scenes` is a user provided `list[str]` --> use it
`select_scenes` is a user provided `list[int]` --> use it
`select_scenes` is a user provided `list[union[int, str]]` --> use it
`select_scenes` is default `None` --> generate integer indices of `list[int]`

Reply:
Got it. Ill make those changes today or tomorrow.